source/common/http/http2/conn_pool.cc

```diff
@@ -259,17 +388,60 @@ void ConnPoolImpl::onStreamReset(ActiveClient& client, Http::StreamResetReason r
   }
 }
 
-void ConnPoolImpl::onUpstreamReady() {
-  // Establishes new codec streams for each pending request.
-  while (!pending_requests_.empty()) {
```

Comment:
In the HTTP/2 case I think we can still have multiple pending requests per connection created, right? Don't we still need a loop to deal with one we get a new connection?